lldb/source/Target/Process.cpp

```diff
@@ -3239,6 +3241,14 @@ Status Process::ConnectRemote(llvm::StringRef remote_url) {
   return error;
 }
 
+void Process::SetRunDirection(RunDirection direction) {
+  if (m_base_direction == direction) {
+    return;
+  }
+  m_thread_list.DiscardThreadPlans();
```

Comment:
Somewhere we should say that we're allowing ourselves the simplification that the ThreadPlanStacks have to be homogeneous in direction, but that we should be able to relax that in the future.  Maybe here or maybe there should be a more general comment describing your strategy here.

Reply:
Renamed this to `SetBaseDirection()` and added this to the comment on that method in `Process.h`.